common/thrift_router.cpp

```diff
@@ -105,26 +121,34 @@ std::unique_ptr<const detail::ClusterLayout> parseConfig(std::string content) {
     }
 
     cl->segments[segment].shard_to_hosts.resize(shard_number);
-    // for each host:port:az
-    for (const auto& host_port_az : segment_value.getMemberNames()) {
-      if (host_port_az == SHARD_NUM_STRs[0] ||
-          host_port_az == SHARD_NUM_STRs[1]) {
+    // for each host:port:group
+    for (const auto& host_port_group : segment_value.getMemberNames()) {
+      if (host_port_group == SHARD_NUM_STRs[0] ||
+          host_port_group == SHARD_NUM_STRs[1]) {
         continue;
       }
 
       detail::Host host;
-      if (!parseHost(host_port_az, &host)) {
-        LOG(ERROR) << "Invalid host port az " << host_port_az;
+      if (!parseHost(host_port_group, &host, segment, local_group)) {
+        LOG(ERROR) << "Invalid host port group " << host_port_group;
         return nullptr;
       }
 
-      const detail::Host* pHost = &*(cl->all_hosts.insert(host).first);
-      const auto& shard_list = segment_value[host_port_az];
+      if (cl->all_hosts.find(host.addr) == cl->all_hosts.end()) {
```

Comment:
we can do cl->all_hosts.find(host) instead of find(host.addr). So we don't have to change the map to set above.

Reply:
the reason is that i need to update the element in the set in place, the set::find() returns const iterator.

Reply:
Here is what i tried:

set<Host>::iterator it = cl->all_hosts.find(host);
it-> groups_prefix_lengths.insert(host. groups_prefix_lengths.begin(), host. groups_prefix_lengths.end())

gcc returns error that cannot manipulate a const iterator

Reply:
Can we get around it by inserting existing lengths into "host", and replace the current one with "host" by "erase()" and "insert()".

Reply:
sounds good